test/jasmine/assets/drag.js

```diff
@@ -1,42 +1,146 @@
 var isNumeric = require('fast-isnumeric');
+
 var mouseEvent = require('./mouse_event');
+var touchEvent = require('./touch_event');
 var getNodeCoords = require('./get_node_coords');
 var delay = require('./delay');
 
-function makeFns(node, dx, dy, opts) {
+/**
+ * Inside `opts`:
+ *
+ * @param {array of 2 numbers} pos0 :
+ *  px position of start of drag motion, default computed using `node` and `edge`
+ * @param {DOM element} node :
+ *  element to drag on, default found using `pos0`
+ * @param {string} edge :
+ *  combinations of 'n', 's', 'e', 'w' used to find `pos0` of `node`
+ *
+ * Set one of:
+ * @param {array of arrays of numbers} path :
+ *  px position drag path
+ * @param {array of 2 numbers} dpos :
+ *  px position delta
+ * @param {array of 2 numbers} posN :
+ *  px position of end of drag motion
+ *
+ * If using `dpos` or `posN`
+ * @param {number} nsteps :
+ *  set number of steps to take between `pos0` and `pos0` + `dpos` or `posN`, default is 1
+ *
+ * @param {boolean} touch :
+ *  pass `true` to simulate touch events
+ * @param {boolean} shiftKey, altKey, ctrlKey ....
+ * pass `true to simulate <shift>, alt, ctrl drag (see ./mouse_event.js for more info)
+ *
+ * @param {function} clearThrottle :
+ *  pass Lib.clearThrottle to clear throttle for all mouse/touch event
+ * @param {boolean} noCover :
+ *  do not wait for "drag cover" element to start "move" events
+ *
+ * @return {object}
+ *  - {function} start
+ *  - {function} end
+ */
+function makeFns(opts) {
```

Comment:
We can now use the same `drag` helpers by passing a path, start/end position, start position + delta, etc..